ddtrace/internal/pack.h

```diff
@@ -41,13 +41,10 @@ static inline int msgpack_pack_write(msgpack_packer* pk, const char *data, size_
     size_t bs = pk->buf_size;
     size_t len = pk->length;
 
+    // return an error if we are going to overflow the buffer, no resizing
+    // this generally means a single trace is larger than the buffer size
     if (len + l > bs) {
-        bs = (len + l) * 2;
-        buf = (char*)PyMem_Realloc(buf, bs);
-        if (!buf) {
-            PyErr_NoMemory();
-            return -1;
-        }
```

Comment:
The nice thing about this is we can choose a max buffer size, and we'll never exceed this memory limit.

Yes, when we recreate a new Packer (per-call to `encode_trace`) we'll get rid of the buffer, but if we have a 20mb trace, we'll still end up allocating a 20mb buffer for encoding when it is not needed.